homeassistant/components/samsungtv/bridge.py

```diff
@@ -34,13 +37,44 @@
 )
 
 
+def mac_from_device_info(info):
+    """Extract the mac address from the device info."""
+    dev_info = info.get("device", {})
+    if dev_info.get("networkType") == "wireless" and dev_info.get("wifiMac"):
+        return format_mac(dev_info.get("wifiMac"))
+    return None
+
+
+async def async_get_device_info(hass, bridge, host):
+    """Fetch the port, method, and device info."""
+    return await hass.async_add_executor_job(_get_device_info, bridge, host)
+
+
+def _get_device_info(bridge, host):
```

Comment:
Side note: The bridge handling would really benefit from type annotations.

Reply:
I agree. I thought the same when I was working on this.